src/meshes/mesh_io.jl

```diff
@@ -225,11 +225,88 @@ function save_mesh_file(mesh::P4estMesh, output_directory, timestep, mpi_paralle
     return filename
 end
 
-# TODO: Implement this function as soon as there is support for this in `t8code`.
-function save_mesh_file(mesh::T8codeMesh, output_directory, timestep, mpi_parallel)
-    error("Mesh file output not supported yet for `T8codeMesh`.")
+# This routine works for both, serial and MPI parallel mode. The forest
+# information is collected on all ranks and then gathered by the root rank.
+# Since only the `levels` array of unsigned bytes is basicially independent per
+# rank it is not worth the effort to have a collective write to the HDF5 file.
```

Comment:
Out of curiosity: This mean that all other data is present and synchronized on all ranks?

Reply:
No, the meaning behind this comment is, that only the `levels` array together with the `num_elements_per_tree` array is sufficient in order to reconstruct the whole forest. Since both arrays are not extremely data intensive, I decided against a collective write from all processes and instead simply gather everything on the root rank.

Reply:
Made the comment clearer.